### PR TITLE
command interpolation for dummies

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

File: blog/_posts/28-04-2017-command-interpolation-for-dummies.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Just make it look like a command and it will run like a command! 
+
+But this falls apart as soon as we start interpolating into the command.
+Let's consider the following:
+
+```Julia
+julia>flag1 = "-e 'println(\"hello world\")'"
+julia>flag2 = "println(\"hello world\")"
+julia>helloworld = "hello world"
+julia>`julia $flag1
```
Comment (omus) [inline blog/_posts/28-04-2017-command-interpolation-for-dummies.md]: Missing ending backtick

File: blog/_posts/28-04-2017-command-interpolation-for-dummies.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+```
+If this all makes sense to you, welcome to the group of enlightened people!
+I looked more like this:
+
+![image](https://cloud.githubusercontent.com/assets/1010467/25554703/66a7ee96-2cd4-11e7-816b-496fcef9837e.png)
+
+
+Obviously this is all well thought out and what was missing is the right mental model.
+
+First of all, keep in mind that there is always a main sitting somewhere, patiently waiting for you to feed it an array of strings.
```
Comment (Sacha0) [inline blog/_posts/28-04-2017-command-interpolation-for-dummies.md]: Perhaps code-quote `main`?

**Comment by ViralBShah** [comment]
Any further thoughts here?

**Comment by SimonDanisch** [comment]
Doesn't seem so!

**Comment by ViralBShah** [comment]
Let's merge in a day or so unless there any other input or objections.

File: blog/_posts/28-04-2017-command-interpolation-for-dummies.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Since this looks exactly like what I would type into the terminal, I inferred that this is the general rule:
+Just make it look like a command and it will run like a command! 
+
+But this falls apart as soon as we start interpolating into the command.
+Let's consider the following:
+
+```Julia
+julia>flag1 = "-e 'println(\"hello world\")'"
+julia>flag2 = "println(\"hello world\")"
+julia>helloworld = "hello world"
```
Comment (omus) [inline blog/_posts/28-04-2017-command-interpolation-for-dummies.md]: Should be a space between the `julia>` prompt and the code.

Comment (omus) [inline blog/_posts/28-04-2017-command-interpolation-for-dummies.md]: The lines are suppose to be silent should have `;` at the end of the line

Comment (ararslan) [inline blog/_posts/28-04-2017-command-interpolation-for-dummies.md]: With a line break between each

File: blog/_posts/28-04-2017-command-interpolation-for-dummies.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+If I understand correctly, all other characters are just conventions.
+So the minus in `-e` isn't treated special, as I assumed at some point.
+And therefore the `-e` and `'println("hello world")'` only belong together semantically, but not when constructing the array of strings.
+So trying to interpolate `'-e'` together with the `'...'` block failed miserably, since Julia tried to interpret it as one string. 
+What made me fail to see the correct solution was, that Julia can actually do better than the shell, 
+since we can directly pass an array of strings to the command.
+So this actually works: 
+```Julia
+run(Cmd(["julia", "-e", "println(\"hello world\")"]))
+```
```
Comment (GlenHertz) [inline blog/_posts/28-04-2017-command-interpolation-for-dummies.md]: Nice post.  You may want to add when arrays are interpolated into \`cmd\` objects, they are handled as multiple arguments to main:

```Julia
args = ["-e", """println("hello world")"""]
`julia $args`
```

When arrays are interpolated into a Cmd they are treated as separate args.  I see no need going to `Cmd()` for that

File: blog/_posts/28-04-2017-command-interpolation-for-dummies.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+So we can in fact have better interpolation, since we just need to insert a new string into the array.
+Which is why this actually works:
+`julia -e $("println(\"hello world\"))`
+Just leave out any `'...'` and let Julia turn it into a command with correctly escaped strings.
+
+Now, Keno Fischer suggested a way of printing that makes the array nature of a command much clearer:
+
+![image](https://cloud.githubusercontent.com/assets/1010467/25554611/afa146c2-2cd1-11e7-93ac-cfc851b67133.png)
```
Comment (omus) [inline blog/_posts/28-04-2017-command-interpolation-for-dummies.md]: This was added in Julia 0.7

**Comment by ViralBShah** [comment]
Maybe update this to publish?

**Comment by SimonDanisch** [comment]
Sounds good - I added updates for the comments!

**Comment by SimonDanisch** [comment]
bump

**Comment by ViralBShah** [comment]
Needs to be updated to today's date and merged.

File: blog/_posts/14-09-2017-command-interpolation-for-dummies.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+So the minus in `-e` isn't treated special, as I assumed at some point.
+And therefore the `-e` and `'println("hello world")'` only belong together semantically, but not when constructing the array of strings.
+So trying to interpolate `'-e'` together with the `'...'` block failed miserably, since Julia tried to interpret it as one string.
+What made me fail to see the correct solution was, that Julia can actually do better than the shell,
+since we can directly pass an array of strings to the command.
+So this actually works:
+```Julia
+run(Cmd(["julia", "-e", "println(\"hello world\")"]))
+```
+Not the above is equivalent to:
```
Comment (ViralBShah) [inline blog/_posts/14-09-2017-command-interpolation-for-dummies.md]: Perhaps you mean Now, instead of Not?

**Comment by SimonDanisch** [comment]
Sorry, didn't see that changes where requested!

**Pull request merged**